vtm/src/org/oscim/renderer/GLUtils.java

```diff
@@ -107,26 +109,89 @@ public static int loadTexture(byte[] pixel, int width, int height, int format,
         return textureIds[0];
     }
 
-    public static void checkGlError(String op) {
-        //GL = GLAdapter.get();
+    /**
+     * Get GL error
+     * See: https://www.khronos.org/opengl/wiki/OpenGL_Error
+     *
+     * @param op the operation which should be debugged
+     * @return the OpenGL error codes
+     */
+    public static List<Integer> checkGlError(String op) {
 
-        int error;
-        while ((error = gl.getError()) != 0) { // GL20.NO_ERROR) {
-            log.error(op + ": glError " + error);
+        int error; // GL.NO_ERROR
+        List<Integer> errors = new ArrayList<>();
+        while ((error = gl.getError()) != GL.NO_ERROR) {
+            log.error(op + ": \tglError " + getGlErrorString(error) + " (" + error + ")");
+            errors.add(error);
             // throw new RuntimeException(op + ": glError " + error);
         }
+        return errors;
```

Comment:
Better use heavier implementations with list creation when needed.
And keep the simpler one for all other frequent cases.

Reply:
Good idea :+1: